src/mapgen_functions.cpp

```diff
@@ -867,6 +867,17 @@ void mapgen_road( mapgendata &dat )
         }
     }
 
+    // place street lights
+    if( neighbor_sidewalks ) {
+        for( int x = 0; x < SEEX * 2 - 1; x++ ) {
+            for( int y = 0; y < SEEY * 2 - 1; y++ ) {
+                if( x % 5 == 0 && y % 5 == 0 && m->ter( point( x, y ) ) == t_sidewalk ) {
+                    m->furn_set( point( x, y ), f_street_light );
```

Comment:
Could I suggest against expanding the big list of global furniture/terrain ids and using the STATIC macro here?
```suggestion
                    m->furn_set( point( x, y ), STATIC( std::string( "f_street_light" ) ) );
```
This might also work, but I'm not sure, I'd have to review what the macro says/does.
```suggestion
                    m->furn_set( point( x, y ), STATIC( "f_street_light" ) );
```

I'd like to do this because having those ids like that makes the game throw an error when you load it without those, such as when running the game with `dda` disabled.

Reply:
I'm not sure how this should work at all. I've tried both 
`m->furn_set( point( x, y ), STATIC( std::string( "f_street_light" ) )`
and
`m->furn_set( point( x, y ), STATIC( "f_street_light" ) )`
and both threw an error like `furn_set` can't use const CachedType as second argument.

Reply:
Probably I have misunderstood how the macro works. I will look at this more.